redisson-tomcat/redisson-tomcat-8/src/main/java/org/redisson/tomcat/UsageValve.java

```diff
@@ -22,6 +22,7 @@
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpSession;
 import java.io.IOException;
+import java.util.Optional;
```

Comment:
In case of UpdateMode.AFTER_REQUEST and readMode == ReadMode.REDIS we need to add both UpdateValve and UsageValve

Reply:
true, it looks like right now we only add one or another
https://github.com/redisson/redisson/blob/master/redisson-tomcat/redisson-tomcat-8/src/main/java/org/redisson/tomcat/RedissonSessionManager.java#L257-L263